File: 283_moveZeroes/sol.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        zeroes = 0
        non_zeroes = []

        for num in nums:
            if num == 0:
                zeroes += 1
            else:
                non_zeroes.append(num)
        
        non_zeroes = non_zeroes + [0] * zeroes

        for i in range(len(non_zeroes)):
            nums[i] = non_zeroes[i]
        

File: 283_moveZeroes/sol.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int zeroes = 0;
        vector<int> non_zeroes;
        
        for (int num : nums) {
            if (num == 0) {
                zeroes++;
            } else {
                non_zeroes.push_back(num);
            }
        }
        
        for (int i = 0; i < zeroes; i++) {
            non_zeroes.push_back(0);
        }
        
        for (int i = 0; i < non_zeroes.size(); i++) {
            nums[i] = non_zeroes[i];
        }
    }
};